makefiles/libc/newlib.mk

```diff
@@ -45,29 +54,32 @@ NEWLIB_INCLUDE_PATTERNS ?= \
 # the patterns above. We use the -isystem gcc/clang argument to add the include
 # directories as system include directories, which means they will not be
 # searched until after all the project specific include directories (-I/path)
-NEWLIB_INCLUDE_DIR ?= $(firstword $(wildcard $(NEWLIB_INCLUDE_PATTERNS)))
+NEWLIB_INCLUDE_FILE = $(firstword $(wildcard $(addsuffix /newlib.h,$(NEWLIB_INCLUDE_PATTERNS))))
 
 # If nothing was found we will try to fall back to searching for a cross-gcc in
 # the current PATH and use a relative path for the includes
-ifeq (,$(NEWLIB_INCLUDE_DIR))
+ifeq (,$(NEWLIB_INCLUDE_FILE))
+  $(warning Using relative include dir)
   NEWLIB_INCLUDE_DIR := $(abspath $(wildcard $(dir $(shell command -v $(PREFIX)gcc 2>/dev/null))/../$(TARGET_ARCH)/include))
+else
+  NEWLIB_INCLUDE_DIR := $(dir $(NEWLIB_INCLUDE_FILE))
 endif
 
-ifeq ($(TOOLCHAIN),llvm)
+ifneq (1,$(USE_NANO_SPECS_FILE))
   # A cross GCC already knows the target libc include path (build-in at compile time)
   # but Clang, when cross-compiling, needs to be told where to find the headers
   # for the system being built.
-  # We also add -nostdinc to avoid including the host system headers by mistake
-  # in case some header is missing from the cross tool chain
-  NEWLIB_INCLUDES := -isystem $(NEWLIB_INCLUDE_DIR) -nostdinc
+  NEWLIB_INCLUDES := -isystem $(NEWLIB_INCLUDE_DIR)
   NEWLIB_INCLUDES += $(addprefix -isystem ,$(abspath $(wildcard $(dir $(NEWLIB_INCLUDE_DIR))/usr/include)))
-endif
-
-ifeq (1,$(USE_NEWLIB_NANO))
-  NEWLIB_NANO_INCLUDE_DIR ?= $(NEWLIB_INCLUDE_DIR)/newlib-nano
-  # newlib-nano overrides newlib.h and its include dir should therefore go before
-  # the regular system include dirs.
-  INCLUDES := -isystem $(NEWLIB_NANO_INCLUDE_DIR) $(INCLUDES)
+  ifeq (1,$(USE_NEWLIB_NANO))
+    # newlib-nano include directory is called either newlib-nano or nano. Use the one we find first.
+    NEWLIB_NANO_INCLUDE_DIR ?= $(firstword $(wildcard $(addprefix $(NEWLIB_INCLUDE_DIR)/, newlib-nano nano)))
+    ifneq ($(NEWLIB_NANO_INCLUDE_DIR),)
+      # newlib-nano overrides newlib.h and its include dir should therefore go before
+      # the regular system include dirs, but after the other newlib includes.
+      NEWLIB_INCLUDES += -isystem $(NEWLIB_NANO_INCLUDE_DIR)
```

Comment:
Why is the nano include dir after the regular newlib dir? The comment does not make sense